parquet-hadoop/src/main/java/org/apache/parquet/filter2/dictionarylevel/DictionaryFilter.java

```diff
@@ -363,14 +363,54 @@ public Boolean visit(Not not) {
         "This predicate contains a not! Did you forget to run this predicate through LogicalInverseRewriter? " + not);
   }
 
+  private <T extends Comparable<T>, U extends UserDefinedPredicate<T>> Boolean visit(UserDefined<T, U> ud, boolean inverted) {
+    Column<T> filterColumn = ud.getColumn();
+    ColumnChunkMetaData meta = getColumnChunk(filterColumn.getColumnPath());
+
+    // Cannot make assumptions around UDPs and acceptance of null values
+    if (meta == null || hasNonDictionaryPages(meta)) {
```

Comment:
When `meta == null`, the column is missing and all values in that column are null. In @viirya's recent patch, 71cff7c5940b7101ff098601850d46b7a4698180, the user-defined predicate is called with null to see whether all rows can be filtered. I think this patch can work the same way.